client/src/app/submission/metadata-list/metadata-list.component.ts

```diff
@@ -48,37 +45,44 @@ export class MetadataListComponent implements OnInit, OnDestroy {
     Object.assign(this._config, config);
   }
 
-  metadataList$: Observable<PagedData<MetadataDocument>>;
   @Input() submissionEnvelopeId: string;
   page: Page = {number: 0, size: 0, sort: '', totalElements: 0, totalPages: 0};
   rows: any[];
   expandAll: boolean;
   isPaginated: boolean;
   validationStates: string[];
   filterState: string;
-  currentPageInfo: {};
-  private alive: boolean;
-  private pollInterval: number;
-  isLoading = false;
 
   constructor(private ingestService: IngestService,
               private flattenService: FlattenService,
               private schemaService: SchemaService,
               private loaderService: LoaderService,
               public dialog: MatDialog) {
-    this.pollInterval = 5000; // 5s
-    this.alive = true;
     this.page.number = 0;
     this.page.size = 20;
-    this.pollingTimer = timer(0, this.pollInterval).pipe(takeWhile(() => this.alive)); // only fires when component is alive
     this.validationStates = ['Draft', 'Validating', 'Valid', 'Invalid'];
   }
 
   ngOnDestroy() {
-    this.alive = false; // switches your TimerObservable off
+    this.dataSource.disconnect();
   }
 
   ngOnInit() {
+    this.dataSource.connect(true, 5000).subscribe(data => {
+      this.rows = data.data.map(row => this.flattenService.flatten(row));
+      this.metadataList = data.data;
+      if (data.page) {
+        this.isPaginated = true;
+        this.page = data.page;
+      } else {
+        this.isPaginated = false;
+      }
+    });
+
+    if (this.dataSource.resourceType === 'files') {
+      this.validationStates = this.validationStates.concat(INVALID_FILE_TYPES);
```

Comment:
Interesting approach, I thought there will be a secondary dropdown when the filter state is invalid. But I think this makes sense and is simpler. Although, the users might not easily see that they are able to filter by these file error types now.

Reply:
Yeah after discussion with gabs, this is what we came up with